File: Discrete Mathematics/Basics_Of_Digital_Logic_Designs/Java_Implementation/NOT_gate.java
public class NOT_gate {
    public String is_NOT_gate(int p){

        if (p == 0 || p == 1){

            if (p == 1){
                return "0";
            }
            else{
                return "1";
            }
        }

        else{
            System.out.println("Invalid input \n Input must be 0 or 1...");
        }
        return "";
    }


    // Test
    String output;
    public void main(String[] args) {
        output =  is_NOT_gate(0);
        System.out.println(output); // output: 1

        output =  is_NOT_gate(1);
        System.out.println(output); // output: 0
    }
}

File: Discrete Mathematics/Basics_Of_Digital_Logic_Designs/C++ Implementation/src/NOT_Gate.cpp
#include "NOT_Gate.h"

int NOT_Gate::is_NOT_Gate(int x)
{
    if (x >=0 && x <=1)
    {
        if(x==1)
        {
            return 0;
        }
        else
        {
            return 1;
        }
    }
    else
    {
        cout<< "Inputs must be 0 or 1 \n";
    }
}
